src/os/bluestore/KernelDevice.cc

```diff
@@ -749,7 +763,10 @@ void KernelDevice::_aio_log_finish(
 	   << std::hex << offset << "~" << length << std::dec << dendl;
   if (cct->_conf->bdev_debug_inflight_ios) {
     std::lock_guard l(debug_lock);
-    debug_inflight.erase(offset, length);
+    if (debug_inflight_write.intersects(offset, length))
+      debug_inflight_write.erase(offset, length);
+    if (debug_inflight_read.intersects(offset, length))
```

Comment:
you might want to use "else if" here - no need to attempt removal from debug_inflight_read if debug_inflight_write had the range.